core/azure-core-amqp/src/main/java/com/azure/core/amqp/exception/ErrorContext.java

```diff
@@ -2,65 +2,59 @@
 // Licensed under the MIT License.
 package com.azure.core.amqp.exception;
 
+import com.azure.core.amqp.AmqpConnection;
+import com.azure.core.amqp.AmqpLink;
+import com.azure.core.amqp.AmqpSession;
 import com.azure.core.implementation.util.ImplUtils;
 
 import java.io.Serializable;
 import java.util.Locale;
-import java.util.Objects;
 
 /**
- * Provides context for an {@link AmqpException}.
+ * Provides context for an {@link AmqpException} that occurs in an {@link AmqpConnection}, {@link AmqpSession},
+ * or {@link AmqpLink}.
  *
  * @see AmqpException
+ * @see SessionErrorContext
+ * @see LinkErrorContext
  */
 public class ErrorContext implements Serializable {
+    static final String MESSAGE_PARAMETER_DELIMITER = ", ";
+
     private static final long serialVersionUID = -2819764407122954922L;
 
     private final String namespaceName;
-    private final Throwable exception;
 
     /**
      * Creates a new instance with the provided {@code namespaceName}.
      *
-     * @param exception Exception that caused this error.
-     * @param namespaceName Event Hub namespace of the error context.
-     * @throws NullPointerException when {@code exception} is {@code null}.
+     * @param namespaceName The service namespace of the error.
      * @throws IllegalArgumentException when {@code namespaceName} is {@code null} or empty.
      */
-    public ErrorContext(final Throwable exception, final String namespaceName) {
-        Objects.requireNonNull(exception);
-
+    public ErrorContext(String namespaceName) {
```

Comment:
Is an ErrorContext just a wrapper around a namespaceName? Do you imagine there will be more information included here over time, or should the exceptions just be extended to include a namespaceName, rather than an ErrorContext?

Reply:
Nevermind, I see subclasses of ErrorContext further down....